next-docs/pages/templates/typescript/contracts.mdx

```diff
@@ -1 +1,453 @@
-# Contracts
+# Contracts (Vanilla and React)
+
+This section explains the contracts for the vanilla and react templates.
+The Three.JS template uses diffeerent contracts.
+
+The onchain components can be divided into two types of functionality:
+
+- **Data**, the part of the system that stores the information.
+- **Logic**, the part of the system that can be called to read or modify the data.
+
+## Data
+
+### The data schema
+
+#### `mud.config.ts`
+
+The data schema is declared in `packages/contracts/mud.config.ts`.
+[Read more details about the schema definition here](/world/config#global-configuration-keys).
+
+The data schema provided in the example is extremely simple (one singleton).
+To see more of how it works, lets add a second table:
+
+```ts filename="mud.config.ts" copy
+import { mudConfig } from "@latticexyz/world/register";
+
+export default mudConfig({
+  tables: {
+    Counter: {
+      keySchema: {},
+      valueSchema: "uint32",
+    },
+  },
+});
+```
+
+There are two automatically generated files related to the data:
+
+- `packages/contracts/src/codegen/index.sol`
+- `packages/contracts/src/codegen/tables/Counter.sol`
+
+<details>
+
+<summary>The automatically generated data files</summary>
+
+`index.sol`
+
+This file just imports all of the automatically generated tables and their identifiers.
+
+```solidity filename="index.sol"
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+/* Autogenerated file. Do not edit manually. */
+
+import { Counter, CounterTableId } from "./tables/Counter.sol";
+```
+
+In this case there is only one table, `Counter`.
+
+`Counter.sol`
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
```

Comment:
nit (does it actually generate 0.8.0? if so there might be an issue with the codegen script)
```suggestion
pragma solidity >=0.8.21;
```

Reply:
You're right. My bad - I updated most of the code, but forgot that line.